src/envoy/http/authn/authenticator_base.cc

```diff
@@ -43,7 +43,7 @@ bool isRuleMatchedWithPeer(const iaapi::CredentialRule& rule,
 AuthenticatorBase::AuthenticatorBase(
     FilterContext* filter_context,
     const AuthenticatorBase::DoneCallback& done_callback)
-    : filter_context_(&filter_context), done_callback_(done_callback) {}
+    : filter_context_(*filter_context), done_callback_(done_callback) {}
```

Comment:
OK,  if you want to use raw pointer as it could be modified,  please put it as last argument (per google c++ style).